src/main/java/io/fabric8/maven/docker/BuildMojo.java

```diff
@@ -71,7 +77,10 @@ protected void buildAndTag(ServiceHub hub, ImageConfiguration imageConfig)
         ImagePullManager pullManager = getImagePullManager(determinePullPolicy(imageConfig.getBuildConfiguration()), autoPull);
         BuildService buildService = hub.getBuildService();
 
-        buildService.buildImage(imageConfig, pullManager, buildContext);
+        File buildArchiveFile = buildService.buildArchive(imageConfig, buildContext, buildArchiveOnly);
```

Comment:
as mentioned below, the option should be already resolved here to the path (if any) for calling `buildArchive`